airbyte-cdk/python/airbyte_cdk/utils/traced_exception.py

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+import traceback
+from datetime import datetime
+
+from airbyte_cdk.models import AirbyteErrorTraceMessage, AirbyteMessage, AirbyteTraceMessage, FailureType, TraceType
+from airbyte_cdk.models import Type as MessageType
+
+
+class AirbyteTracedException(Exception):
+    """
+    TODO: give this a meaningful description
+    """
+
+    def __init__(
+        self,
+        internal_message: str = None,
+        message: str = None,
+        failure_type: FailureType = FailureType.system_error,
+        exception: BaseException = None,
+    ):
+        self.internal_message = internal_message
+        self.message = message
+        self.failure_type = failure_type
+        self._exception = exception
+        super().__init__(internal_message)
+
+    def as_airbyte_message(self) -> AirbyteMessage:
+        now_millis = int(datetime.now().timestamp() * 1000)
+
+        trace_exc = self._exception or self
+        stack_trace_str = "".join(traceback.TracebackException.from_exception(trace_exc).format())
+
+        trace_message = AirbyteTraceMessage(
+            type=TraceType.ERROR,
+            emitted_at=now_millis,
+            error=AirbyteErrorTraceMessage(
+                message=self.message or "an error occurred with the source",
```

Comment:
any suggestions on what this generic default message should be are welcome 😄

Reply:
this isn't necessarily source, could also be destination.

How about `Something went wrong in the connector. See the logs for more details.` ?
Pointing them at the logs seems sensible given that this is always going to be non-specific at this level.

Reply:
Are there any environment variables or system arguments we can use to know if we are a source or destination?  e.g. looking at the docker container name (source-mailchimp)?